app/models/meter_reading.rb

```diff
@@ -0,0 +1,39 @@
+class MeterReading < ApplicationRecord
+  belongs_to :billing_contract, optional: true
+
+  validates :value, presence: true, numericality: { only_integer: true }
+  validate :value_bigger_than_previous_one?
+
+  private
+
+  def value_bigger_than_previous_one?
+    if previous_meter_reading
+      errors.add :value, "Must be bigger than previous one" if value < previous_meter_reading
+    else
+      true
+    end
+  end
+
+  def previous_meter_reading
+    billing_contract&.meter_readings&.order(created_at: :desc)&.first&.value
```

Comment:
too many safe navigation

return nil if there is no billing_contract or any meter_readings

Reply:
done